extensions/amp-subscriptions/0.1/crypto-handler.js

```diff
@@ -67,27 +75,38 @@ export class CryptoHandler {
    * @return {!Promise}
    */
   tryToDecryptDocument(decryptedDocumentKey) {
-    if (this.decryptionPromise_) {
-      return this.decryptionPromise_;
+    if (!this.shaKeyHash_) {
+      return Promise.reject(new Error('Missing Document Key Hash'));
```

Comment:
Are we saying the publisher also has to include the hash in the non-cache version of the doc?   I thought we only cared if the doc was being served from a cache.

Reply:
True, that's what the original PR was doing. I changed it so that the decryption only fails if the hashes are unequal, not if it is not found in the head.